File: Roadmap/02 - FUNCIONES Y ALCANCE/scala/rianojnicolas.scala
import scala.util.Random

object Main {
    def main(args: Array[String]): Unit = {
        // Variable Global
        var varGlobal = "Muchos Exitos, y que tenas un buen día"

        // Funciones:
        // Funcion sin parametros ni retorno
        def welcomePlayer(): Unit = {
            println("""
        Hola People !!, este es un juego improvisado de
        piedra, papel o tijera.
        
        las reglas son las siguientes:
        """)
        // Funcion dentro de otra funcion
            def reglas(): Unit = {
                print("""
            1. Debes ingresar el objeto con el que jugaras
            2. El pc elegira aleatoriamente un objeto
            3. Recuerda que: papel le gana a piedra,
                            piedra le gana a tijera,
                            tijera le gana a papel.
            
            Muchos Exitooooss !!!
            """)
            }
            reglas()
        }


        // Funcion sin parametros con retorno   
        // Funcion para pedir el objeto a jugar
        def functionInput(): String = {
            println("Ingresa tu objeto con el que vas a jugar: ")
            var option = scala.io.StdIn.readLine()
            while(option != "piedra" && option != "tijera" && option != "papel"){
                println("Ingresa tu objeto con el que vas a jugar: ")
                option = scala.io.StdIn.readLine()
            }
            return option
        }


        // (Funcion con un parametro y sin retorno)
        // Funcion competencia
        def playCompetition(optionPlayer:String): Unit = {
            // Concepto de Variable Local
            val optionList = List("piedra", "papel", "tijera")
            var optionPC = Random.shuffle(optionList).head
            println(s"la opcion del pc fue: $optionPC \n")
            if (optionPC == optionPlayer) {
                println("Emptados")
            }
            else if (optionPC == "papel") {
                if (optionPlayer == "piedra") {
                    println("perdiste")
                }
                else if(optionPlayer == "tijera") {
                    println("ganaste")
                }
            }
            else if (optionPC == "tijera") {
                if (optionPlayer == "papel") {
                    println("perdiste")
                }
                else if (optionPlayer == "piedra") {
                    println("ganaste")
                }
            }
            else if (optionPC == "piedra") {
                if (optionPlayer == "papel") {
                    println("ganaste")
                }
                else if (optionPlayer == "tijera") {
                    println("perdiste")
                }
            }
        }

        def run(): Unit = {
            welcomePlayer()
            println(varGlobal)
            var option_player = functionInput()
            playCompetition(option_player)
        }
        
        run()

        /*
        DIFICULTAD EXTRA
        */
        println("*---------------------------------*")
        println("*       DIFICULTAD EXTRA          *")
        println("*---------------------------------*")

        def extraDifficult(param1: String, param2: String): Float = {
            var contador = 0
            for (i <- 0 to 101) {
                if (i%3 == 0 && i%5 == 0) {
                    println(s"para el numero $i: $param1 $param2")
                }
                else if(i%3 == 0) {
                    println(s"para el numero $i: $param1")
                }
                else if(i%5 == 0) {
                    println(s"para el numero $i: $param2")
                }
                else {
                    contador += 1
                    println(s"para el numero $i no hay texto")
                }
            }
            return contador
        }

        println("\nAhora vamos con el ejercicio extra\n")
        println(extraDifficult("hola", "mundo"))
    }
}

File: Roadmap/01 - OPERADORES Y ESTRUCTURAS DE CONTROL/javascript/Roadmap/02 - FUNCIONES Y ALCANCE/python/rianojnicolas.py
import random

# Variable global y funcion ya creada por el lenguaje
variableGlobal = "Paso por aca, a desearte muchos exitos!!!\n"
variableGlobal = variableGlobal.upper()

# (Funcion sin parametros ni retorno)
# Funcion de bienvenida
def welcomePlayer():
    print("""
        Hola People !!, este es un juego improvisado de
        piedra, papel o tijera.
        
        las reglas son las siguientes:
        """)
    # Funcion dentro de otra funcion
    def reglas():
        print("""
            1. Debes ingresar el objeto con el que jugaras
            2. El pc elegira aleatoriamente un objeto
            3. Recuerda que: papel le gana a piedra,
                            piedra le gana a tijera,
                            tijera le gana a papel.
            
            Muchos Exitooooss !!!
            """)
    reglas()


# Funcion sin parametros con retorno
# Funcion para pedir el objeto a jugar
def functionInput():
    option = input("Ingresa tu objeto con el que vas a jugar: ")
    while(option != "piedra" and option != "tijera" and option != "papel"):
        option = input("Ingresa tu objeto con el que vas a jugar: ")
    return option


# (Funcion con un parametro y sin retorno)
# Funcion competencia
def playCompetition(optionPlayer):
    # Concepto de Variable Local
    optionList = ["piedra", "papel", "tijera"]
    optionPC = random.choice(optionList)
    print(f'la opcion del pc fue: {optionPC}\n')
    if (optionPC == optionPlayer):
        print("Emptados")
    elif (optionPC == "papel"):
        if (optionPlayer == "piedra"):
            print("perdiste")
        elif(optionPlayer == "tijera"):
            print("ganaste")
    elif(optionPC == "tijera"):
        if (optionPlayer == "papel"):
            print("perdiste")
        elif (optionPlayer == "piedra"):
            print("ganaste")
    elif(optionPC == "piedra"):
        if (optionPlayer == "papel"):
            print("ganaste")
        elif (optionPlayer == "tijera"):
            print("perdiste")


def run():
    welcomePlayer()
    print(variableGlobal)
    option_player = functionInput()
    playCompetition(option_player)


def extraDifficult(param1, param2):
    contador = 0
    for i in range(0,101):
        if (i%3 == 0 and i%5 == 0):
            print(f'para el numero {i}: {param1} {param2}')
        elif(i%3 == 0):
            print(f'para el numero {i}: {param1}')
        elif(i%5 == 0):
            print(f'para el numero {i}: {param2}')
        else:
            contador += 1
            print(f'para el numero {i} no hay texto')
    return contador


if __name__ == '__main__':
    run()
    print("\nAhora vamos con el ejercicio extra\n")
    print(extraDifficult("hola","mundo"))